backup.go

```diff
@@ -155,6 +155,30 @@ func backup(ID string) error {
 	fmt.Printf("Creating backup of %s (%s, %s)\n", conf.Name[1:], conf.Config.Image, conf.ID[:12])
 
 	paths = []string{}
+
+	var imageName = conf.Config.Image
+	if !strings.Contains(imageName, ":") {
+		// No tag on image used to start container, find the real image
+		images, err := cli.ImageList(ctx, types.ImageListOptions{})
+		if err == nil {
```

Comment:
I'd like to suggest to slightly restructure this code and align the happy path to the left edge. Essentially this means inverting this logic and simply checking for `err != nil`, followed by returning the error. This allows you to keep the rest of the code un-indented, which makes it a bit more readable.

You can read more about this here: https://medium.com/@matryer/line-of-sight-in-code-186dd7cdea88

Reply:
That's a sensible strategy to follow. This block is likely not very easy to follow too, so I've broken it out into a function with fast exit paths where appropriate and comments.